### PR TITLE
whitespace tests and some others

### LINKED ISSUE #16: Space tolerance tests
Do we need tests that ensure support for whitespace in all the right places?

Related to https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/273 which decided to keep them in the ABNF.

### PR DESCRIPTION
**Pull request by gregsdennis**
Resolves #16

I haven't pushed the _cts.json_ changes, though I was able to rebuild it locally.  Not sure if that's done automatically yet.  If not, I'll go ahead and push those changes.

Supported whitespace includes

- space
- tab
- carriage return
- newline

These tests check all of those.  ~My implementation doesn't seem to like some of them yet.  Haven't been able to test whether that's my implementation or something about the way I wrote the tests.  Will continue to check.~

### CODE REVIEW

**Comment by gregsdennis** [comment]
My implementation now passes all of these tests.

Side note:  I've also added a "whitespace tolerance" option that allows spaces in some of these places.

**Review by f3ath — APPROVED** [review]

File: tests/whitespace/operators.json [hunk 1]
```diff
@@ -0,0 +1,388 @@
+{
+  "tests": [
+    {
+      "name": "space before ==",
```
Comment (glyn) [inline tests/whitespace/operators.json:4]: This test name doesn't match the test.

**Review by glyn — APPROVED** [review]
Approved with one test name change required. (github wouldn't let me enter a suggestion!)

File: tests/whitespace/operators.json (hunk 1 above)
Comment (gregsdennis) [reply tests/whitespace/operators.json:4]: Copy/paste.  Will fix after I've slept.

**Comment by f3ath** [comment]
> I haven't pushed the cts.json changes, though I was able to rebuild it locally. Not sure if that's done automatically yet. If not, I'll go ahead and push those changes.

It's not automated yet. Please generate the cts.json before merging.
